File: app/src/main/java/com/proxy/shadowsocksr/impl/plugin/proto/SSProtocol.java
package com.proxy.shadowsocksr.impl.plugin.proto;

import java.util.HashMap;

public final class SSProtocol extends AbsProtocol
{
    public SSProtocol(String rmtIP, int rmtPort, int tcpMss, HashMap<String, Object> shareParam)
    {
        super(rmtIP, rmtPort, tcpMss, shareParam);
    }

    @Override public byte[] beforeEncrypt(byte[] data) throws Exception
    {
        return data;
    }

    @Override public byte[] afterDecrypt(byte[] data) throws Exception
    {
        return data;
    }

}

File: jvmVer/src/main/java/com/proxy/shadowsocksr/impl/plugin/proto/SSProtocol.kt
package com.proxy.shadowsocksr.impl.plugin.proto

import java.util.HashMap

class SSProtocol(rmtIP: String, rmtPort: Int, tcpMss: Int, shareParam: HashMap<String, Any>) : AbsProtocol(
        rmtIP, rmtPort, tcpMss, shareParam)
{
    @Throws(Exception::class)
    override fun beforeEncrypt(data: ByteArray): ByteArray
    {
        return data
    }

    @Throws(Exception::class)
    override fun afterDecrypt(data: ByteArray): ByteArray
    {
        return data
    }

}
